firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -84,6 +84,77 @@ public FirebaseFirestore getFirestore() {
     return firestore;
   }
 
+  /** Validates the given document ID and returns the corresponding FieldValue. */
+  private FieldValue validateAndCalculateReferenceValue(FieldPath fieldPath, Object value) {
+    com.google.firebase.firestore.model.FieldPath internalPath = fieldPath.getInternalPath();
+    hardAssert(
+        internalPath.isKeyField(),
+        "fieldPath must be a key field to calculate the reference value.");
```

Comment:
So we don't use `fieldPath` for anything other than to assert that `isKeyField()` is true?  In that case, I think we should just drop the parameter.